lib/elixir/lib/kernel.ex

```diff
@@ -2924,6 +2926,10 @@ defmodule Kernel do
     end
   end
 
+  defp comp(left, {:|, _, [h, t]}) do
+    quote(do: :erlang.or(unquote(left) in unquote(t), :erlang."=:="(unquote(left), unquote(h))))
```

Comment:
Ok, cool. I wasn't sure if we wanted to preserve the exact semantics of the other cases or whether performance was more important. Will make the change.